File: Workout/app/src/main/java/first/example/workout/MainActivity.java
package first.example.workout;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentTransaction;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity implements WorkoutListFragment.Listener{

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public void itemClicked(long id) {
        View fragCont=findViewById(R.id.frag_container);
        if(fragCont!=null) {
WorkoutDetailFragment fr=new WorkoutDetailFragment();
            FragmentTransaction transaction=getSupportFragmentManager().beginTransaction();
            fr.setWorkoutId(id);
            transaction.replace(R.id.frag_container,fr);
            transaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE);
            transaction.addToBackStack(null);
            transaction.commit();
        }
            else{
            Intent intent = new Intent(this, DetailActivity.class);
            intent.putExtra(DetailActivity.EXTRA_WORKOUT_ID, (int) id);
            startActivity(intent);
        }
    }
}

File: TimerMVVM/app/src/main/java/com/stebakov/timermvvm/MainActivity.kt
package com.stebakov.timermvvm

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val textView = findViewById<TextView>(R.id.textView)
        val observable = TextObservable()
        observable.observe(object : TextCallback{
            override fun updateText(str: String) = runOnUiThread {
                textView.text = str
                Log.d("MyLog","updateText at MainActivity, str = $str")
            }
        })
        val viewModel = ViewModel(observable)
        viewModel.init()
    }
}